### PR TITLE
👷‍♂️ simplify the build process

### PR DESCRIPTION
**Pull request by bluebill1049**
## Description

we are no longer doing env check, so we can safely remove those extra builds, and make our build simpler and faster. Maybe we should consider moving to the micro build package which used in the resolver? hopefully, it doesn't blow up the size tho.

### CODE REVIEW

File: scripts/rollup/createRollupConfig.js [hunk 1]
```diff
@@ -56,12 +30,8 @@ export function createRollupConfig(options, callback) {
         commonjs({
           include: /\/node_modules\//,
         }),
-      options.env !== undefined &&
-        replace({
-          'process.env.NODE_ENV': JSON.stringify(options.env),
-        }),
       sourcemaps(),
-      shouldMinify &&
+      options.format !== 'esm' &&
```
Comment (jorisre) [inline scripts/rollup/createRollupConfig.js:34]: I think you can minify for all bundles

**Review by jorisre — APPROVED** [review]
LGTM ! I'll try again microbundle

File: scripts/rollup/createRollupConfig.js (hunk 1 above)
Comment (bluebill1049) [reply scripts/rollup/createRollupConfig.js:34]: Thanks for the review @jorisre i remember last time @kotarella1110 pointed me to a PR that was recommended not minify esm.

Comment (jorisre) [reply scripts/rollup/createRollupConfig.js:34]: Ah ! I want to know why :)
By default, microbundle minify esm.
In all case, when user import React Hook Form esm, the bundler of its project will minify it.

Comment (kotarella1110) [reply scripts/rollup/createRollupConfig.js:34]: @bluebill1049 @jorisre Please see this https://github.com/formium/tsdx/issues/155 👍

Comment (kotarella1110) [reply scripts/rollup/createRollupConfig.js:34]: As joris said, in the case of esm, the project bandler can minify, and the decision to minify should be left to the user.

**Review by kotarella1110 — APPROVED** [review]
LGTM 🎉

**Comment by bluebill1049** [comment]
thanks for the review guys.

**Pull request merged**